### PR TITLE
I'm add MongoDB Extension

### PR DESCRIPTION
**Pull request by SenseiTarzan**
it could be useful to some people who use mongodb rather than mysql which does the job but very slow compared to mongodb only on the linux version

### CODE REVIEW

**Comment by SenseiTarzan** [comment]
Hi

File: compile.sh [hunk 1]
```diff
@@ -924,6 +940,10 @@ RANLIB=$RANLIB CFLAGS="$CFLAGS $FLAGS_LTO" CXXFLAGS="$CXXFLAGS $FLAGS_LTO" LDFLA
 --with-yaml \
 --with-openssl \
 --with-zip \
+--with-libbson="yes" \
+--with-libmongoc="yes" \
+--with-mongodb-system-libs="yes" \
+--with-mongodb-ssl="auto" \
```
Comment (dktapps) [inline compile.sh]: mongodb hasn't been downloaded at this point ...

**Review by dktapps — CHANGES_REQUESTED** [review]
I'm on the fence about adding mongodb, but this PR is clearly broken, and also needs Windows support.

The MacOS build is also broken.

File: compile.sh (hunk 1 above)
Comment (SenseiTarzan) [reply compile.sh]: normally i put the driver on windows it should work and then on mac i found the line that asks but i don't know the rule.
this is line 1072 in compile.sh who create problem for a compilation mac php

**Pull request closed without merging**